app/views/projects/blob/_replace.html.haml

```diff
@@ -0,0 +1,27 @@
+#modal-replace-blob.modal
+  .modal-dialog
+    .modal-content
+      .modal-header
+        %a.close{href: "#", "data-dismiss" => "modal"} ×
+        %h3.page-title Replace #{@blob.name}
+        %p.light
+          From branch
+          %strong= @ref
+      .modal-body
+        = form_tag namespace_project_update_blob_path(@project.namespace, @project, @id), method: :put, id: 'blobFileDropzone', class: 'dropzone form-horizontal' do
+          %br
+            .form-group
+              .col-sm-offset-2.col-sm-10
+                .dropzone-previews
+                  %p.dz-message You can drop a file here or click to upload.
+                  .fallback
+                    = file_field_tag "file", multiple: true, id: "replace_file"
+          %br
+            = render 'shared/commit_message_container', params: params,
+                  placeholder: 'Replace this file because...'
+            .form-group
+              .col-sm-offset-2.col-sm-10
+                = button_tag 'Replace file', class: 'btn btn-small btn-primary btn-replace-file', id: 'submit-all'
+                = link_to "Cancel", '#', class: "btn btn-cancel", "data-dismiss" => "modal"
```

Comment:
Is it necessary to have different templates here? _upload.html.haml and and _replace.html.haml seem identical except for a few types.